File: src/Runner/TestBase.cs
using System;

namespace Runner
{
    public class TestBase
    {
        public const int NumberOfTests = 5;

        private static readonly Lazy<int[]> results;

        static TestBase()
        {
            results = new Lazy<int[]>(() => new int[NumberOfTests]);
        }

        protected void Succeeded(int n)
        {
            results.Value[--n] = 1;
        }

        protected static void TestsEnded()
        {
            var resultString = string.Join(",", results.Value);
            new Communicator().Write(resultString);
        }
    }
}

File: Python/TestBase.py
class TestBase:
	def __init__(self):
		self.results = [0,0,0,0,0]

	def Succeeded(self, n):
		self.results[n - 1] = 1

	def TestEndend(self):
		resultString = ",".join(str(v) for v in self.results)
		f = open("C:\\Users\\roh\\AppData\\Roaming\\DIPS Coding Competition\\results.txt", "w")
		f.write(resultString)
		f.close()